crates/red_knot_workspace/src/site_packages.rs

```diff
@@ -8,15 +8,68 @@
 //! reasonably ask us to type-check code assuming that the code runs
 //! on Linux.)
 
+use std::fmt;
 use std::io;
+use std::num::NonZeroUsize;
+use std::ops::Deref;
 
 use ruff_db::system::{System, SystemPath, SystemPathBuf};
 
-/// Attempt to retrieve the `site-packages` directory
-/// associated with a given Python installation.
+/// The value given by the `home` key in `pyvenv.cfg` files.
+///
+/// This is equivalent to `{sys_prefix_path}/bin`, and points
+/// to a directory in which a Python executable can be found.
+/// Confusingly, it is *not* the same as the [`PYTHONHOME`]
+/// environment variable that Python provides! However, it's
+/// consistent among all mainstream creators of Python virtual
+/// environments (the stdlib Python `venv` module, the third-party
+/// `virtualenv` library, and `uv`), was specified by
+/// [the original PEP adding the `venv` module],
+/// and it's one of the few fields that's read by the Python
+/// standard library's `site.py` module.
+///
+/// Other values, such as the path to the Python executable or the
+/// base-executable `sys.prefix` value, are either only provided in
+/// `pyvenv.cfg` files by some virtual-environment creators,
+/// or are included under different keys depending on which
+/// virtual-environment creation tool you've used.
+///
+/// [`PYTHONHOME`]: https://docs.python.org/3/using/cmdline.html#envvar-PYTHONHOME
+/// [the original PEP adding the `venv` module]: https://peps.python.org/pep-0405/
+#[derive(Debug, PartialEq, Eq)]
+struct PythonHomePath(SystemPathBuf);
+
+impl PythonHomePath {
+    fn new(path: impl AsRef<SystemPath>, system: &dyn System) -> Option<Self> {
```

Comment:
Is it guaranteed that `path` is an absolute path? If not, to what directory is it relative to?

Reply:
uv, virtualenv and the stdlib `venv` module all use absolute paths for the `home` key. And the stdlib `site` module appears to assume that the `home` key will provide an absolute path. But I have no idea if this is specified anywhere or if it's safe to rely on that. Paging @carljm as the person who invented Python virtual environments xD

Reply:
> Paging @carljm as the person who invented Python virtual environments

Not true, that was Ian Bicking! (Or arguably PJ Eby, though his approach didn't catch on and Ian's did.)

> uv, virtualenv and the stdlib `venv` module all use absolute paths for the `home` key. And the stdlib `site` module appears to assume that the `home` key will provide an absolute path. But I have no idea if this is specified anywhere or if it's safe to rely on that.

Sadly this doesn't appear to have been clearly specified in the PEP, and it should have been :/ I think it is safe to rely on, simply because CPython itself won't work correctly with a relative path there (at best it would interpret the relative path as relative to your CWD when you run Python, not as relative to `pyvenv.cfg`, so if it worked at all it would be very fragile based on your CWD), and no venv creator puts a relative path there. But this is definitely something to consider in a potential update of the venv docs.